OSDOCS-8661: Configuring IP failover content audit (#74064)
openshift/openshift-docs: modules/nw-ipfailover-configuring-check-notify-scripts.adoc

[id="nw-ipfailover-configuring-check-notify-scripts_{context}"]
= Configuring check and notify scripts

Keepalived monitors the health of the application by periodically running an optional user-supplied check script. For example, the script can test a web server by issuing a request and verifying the response. As cluster administrator, you can provide an optional notify script, which is called whenever the state changes.

> The check and notify scripts run in the IP failover pod and use the pod file system, not the host file system. However, the IP failover pod makes the host file system available under the `/hosts` mount path. When configuring a check or notify script, you must provide the full path to the script. The recommended approach for providing the scripts is to use a config map.

Reviewer: ```suggestion

The check and notify scripts run in the IP failover pod and use the pod file system, not the host file system. However, the IP failover pod makes the host file system available under the `/hosts` mount path. When configuring a check or notify script, you must provide the full path to the script. The recommended approach for providing the scripts is to use a configuration map.
```
Second paragraph is tucking up into the first without the line space.

> The full path names of the check and notify scripts are added to the Keepalived configuration file, `_/etc/keepalived/keepalived.conf`, which is loaded every time Keepalived starts. The scripts can be added to the pod with a config map as follows.

Reviewer: ```suggestion
The full path names of the check and notify scripts are added to the Keepalived configuration file, `_/etc/keepalived/keepalived.conf`, which is loaded every time Keepalived starts. The scripts can be added to the pod with a configuration map as described in the following methods.
```
Methods might not be the best word here, but we don't end with "as follows" per style guidance.

Author: Thank you! I'm going to use " `ConfigMap`object" here per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references and update any other instances to align with this usage.

*Check script*

When a check script is not provided, a simple default script is run that tests the TCP connection. This default test is suppressed when the monitor port is `0`.

Each IP failover pod manages a Keepalived daemon that manages one or more virtual IP (VIP) addresses on the node where the pod is running. The Keepalived daemon keeps the state of each VIP for that node. A particular VIP on a particular node might be in `master`, `backup`, or `fault` state.

> If the check script returns non-zero, the node will go into the `backup` state, and any VIPs it holds will be reassigned.

Reviewer: ```suggestion
If the check script returns non-zero, the node enters the `backup` state, and any VIPs it holds are reassigned.
```
No future tense
